cpp/include/raft/linalg/detail/eig.cuh

```diff
@@ -91,6 +91,15 @@ void eigDC(raft::resources const& handle,
 #if CUDART_VERSION < 11010
   eigDC_legacy(handle, in, n_rows, n_cols, eig_vectors, eig_vals, stream);
 #else
+
+  // Use a new stream instead of `cudaStreamPerThread` to avoid cusolver bug # 4580093.
+  cudaStream_t stream_new;
+  cudaEvent_t sync_event;
+  RAFT_CUDA_TRY(cudaStreamCreate(&stream_new));
+  RAFT_CUDA_TRY(cudaEventCreate(&sync_event));
```

Comment:
I think, in this case it would be justified to use the [resource::detail::get_cuda_stream_sync_event](https://github.com/rapidsai/raft/blob/branch-24.08/cpp/include/raft/core/resource/detail/stream_sync_event.hpp#L43) instead of manually managing the resource.
This event resource is normally intended for synchronization between the streams in the resource pool, but as long as you don't use the stream pool resource at the same time, it's ok to reuse the event.

You can also use the stream from the stream pool resource, but there is a small problem with it, that raft/rmm would create 16 streams by default instead of one :)

Reply:
Please note that we shouldn't be calling detail APIs in any namdespaces outside of the immediate namespace where those detail APIs reside. Please expose this function if it's going to be used publicly.